commit cb97fd309790330bced04091fac46ea1466283c9
Author: Anatol Belski <ab@php.net>
Date:   Fri Jun 29 18:34:53 2018 +0200

    Fixed bug #76548 pg_fetch_result did not fetch the next row

diff --git a/ext/pgsql/pgsql.c b/ext/pgsql/pgsql.c
index 904b9aa..6da01d4 100644
--- a/ext/pgsql/pgsql.c
+++ b/ext/pgsql/pgsql.c
@@ -2669,6 +2669,7 @@ PHP_FUNCTION(pg_fetch_result)
 		if (pgsql_row >= PQntuples(pgsql_result)) {
 			RETURN_FALSE;
 		}
+		pg_result->row++;
 	} else {
 		if (row < 0 || row >= PQntuples(pgsql_result)) {
 			php_error_docref(NULL, E_WARNING, "Unable to jump to row " ZEND_LONG_FMT " on PostgreSQL result index " ZEND_LONG_FMT,
diff --git a/ext/pgsql/tests/bug76548.phpt b/ext/pgsql/tests/bug76548.phpt
new file mode 100644
index 0000000..3a0b892
--- /dev/null
+++ b/ext/pgsql/tests/bug76548.phpt
@@ -0,0 +1,23 @@
+--TEST--
+Bug #76548 pg_fetch_result did not fetch the next row
+--SKIPIF--
+<?php include("skipif.inc"); ?>
+--FILE--
+<?php
+include('config.inc');
+
+$conn = pg_connect($conn_str);
+
+$result = pg_query($conn, 'SELECT v FROM (VALUES (1), (2), (3)) AS t(v)');
+
+while ($value = pg_fetch_result($result, 0)) {
+  var_dump($value); // should be 1, 2 then 3.
+}
+
+?>
+==DONE==
+--EXPECTF--
+string(1) "1"
+string(1) "2"
+string(1) "3"
+==DONE==
